master/buildbot/db/buildrequests.py

```diff
@@ -64,8 +64,7 @@ def _saSelectQuery(self):
         from_clause = from_clause.join(builder_tbl,
```

Comment:
I think we said we should only join in case of sourcestamp filtering (branch or repository)